packages/server/lib/controllers/auth.controller.ts

```diff
@@ -260,6 +292,79 @@ class AuthController {
             next(error);
         }
     }
+
+    getHostedLogin(req: Request, res: Response, next: NextFunction) {
+        try {
+            const provider = req.query['provider'] as string;
+
+            if (!workos) {
+                errorManager.errRes(res, 'workos_not_configured');
+                return;
+            }
+
+            const body = req.body;
+
+            const oAuthUrl = workos?.userManagement.getAuthorizationUrl({
+                clientId: process.env['WORKOS_CLIENT_ID'] || '',
+                provider,
+                redirectUri: `${getBaseUrl()}/api/v1/login/callback`,
+                state: body ? Buffer.from(JSON.stringify(body)).toString('base64') : ''
+            });
+
+            res.send({ url: oAuthUrl });
+        } catch (err) {
+            next(err);
+        }
+    }
+
+    async loginCallback(req: Request, res: Response, next: NextFunction) {
+        try {
+            const { code, state } = req.query;
+
+            if (!workos) {
+                errorManager.errRes(res, 'workos_not_configured');
+                return;
+            }
+
+            const { user: authorizedUser } = await workos.userManagement.authenticateWithCode({
+                clientId: process.env['WORKOS_CLIENT_ID'] || '',
+                code: code as string
+            });
+
+            let user: User | null = null;
+            const existingUser = await userService.getUserByEmail(authorizedUser.email);
+
+            if (existingUser) {
+                user = existingUser;
+            } else {
+                const name =
+                    authorizedUser.firstName || authorizedUser.lastName
+                        ? `${authorizedUser.firstName || ''} ${authorizedUser.lastName || ''}`
+                        : authorizedUser.email.split('@')[0];
+
+                const accountId = await createAccountIfNotInvited(state as string, name as string);
```

Comment:
should state and name be validated prior to make sure they are acceptable string. That would also avoid having to cast them